### ISSUE TITLE
Image resolution and model export

### ISSUE THREAD

**Issue by PaulineTreyvaud** [lyuwenyu/RT-DETR#267]
Hi !
Congrats and thank your for your work !

I would like to train on full-HD images (1920 x 1080 pixels {however some have different formats}) and test on images with the same size (custom dataset). I changed the resize parameter of the dataloader.yml for both train and validating to 1056x1056, as well as the *eval_spatial_size* parameter (both l43 and l58) of rtdetr_r50vd.yml to keep maximum resolution while avoiding padding.
This doesn't seem to be a problem for the training part, however the onnx export creates an error : 

=======================================================================
1488, in _slow_forward
	result = self.forward(*input, **kwargs)
  File "/rt-detr/tools/../src/zoo/rtdetr/hybrid_encoder.py", line 147, in forward
	q = k = self.with_pos_embed(src, pos_embed)
  File "./rt-detr/tools/../src/zoo/rtdetr/hybrid_encoder.py", line 141, in with_pos_embed
	return tensor if pos_embed is None else tensor + pos_embed

RuntimeError : the size if tensor a (400) must match the size of tensor b (1089) at non-singleton dimension 1

=======================================================================

Am I missing something ? 

For context : I use the pytorch version, and a config file based on rtdetr_r101vd_6x_coco.yml

Thank you!

**Comment by lyuwenyu** [comment]
try to set `l58` as `~` ( None

**Comment by PaulineTreyvaud** [comment]
I did, the outcome is the same :)

**Comment by raitzjm** [comment]
@PaulineTreyvaud  try also changing Line 43 to  ~

**Comment by raitzjm** [comment]
@PaulineTreyvaud did that work for you ? Because I did the same, but I am still having issues when trying to do inference with the ONNX runtime.

**Comment by PaulineTreyvaud** [comment]
@raitzjm what ended up working is to change both l43 and l58 of rtdetr_r50vd.yml, as well as l53 and l54 of export_onnx.py to the expected resolution ((1056, 1056) in my case).
Also, note that the exported model can re-calibrate the output to the original image size if it is given as a parameter (change orig_target_sizes during inference).

**Comment by eypros** [comment]
I was trying to also train on a custom resolution and tried to follow your advice.
I made the following changes in my code:

a) `- {type: Resize, size: [640, 640], }`  to `- {type: Resize, size: [768, 768], }` inside `./include/dataloader.yml` for `train_dataloader` and `val_dataloader`
b) inside `rtdetrv2_r50vd.yml` set  `eval_spatial_size: [768, 768]`
c) I tried without any other changes but the training fails (in contrast with the comments other made) with error in line:
`assert (boxes1[:, 2:] >= boxes1[:, :2]).all()`
which means the bboxes have some errors. Although, with the default dimension (640) this never happened.

I also tried to change line 43 to either ~ or `None`
change `depth_mult: 1` to `depth_mult: None` or `depth_mult: ~` but both fail.
The same with the line 58:
`box_noise_scale: 1.0` to `box_noise_scale: None` or `box_noise_scale: ~`

Are those the lines I was supposed to change? Is there anything else I should change to make it work for other dimensions?
